### DISCUSSION TITLE
Import contents from One Sunbird environment to another Sunbird Environment

### DISCUSSION THREAD

**Discussion by vinod303** [Sunbird-Knowlg/Community#152] [Q&A] [answered]
**Context:**
We have a couple of scenarios where , there is a need to use the contents in One SB environment(eg. DIKSHA)  in another another SB environment(Say in a SB SaaS instance).

Exploring this, i have come across the following import API, but when I tried , i was constantly getting the  504 Gateway timeout error.  
https://documenter.getpostman.com/view/25463377/2s9YC4VDL4#684e13d3-3f2f-4470-8c41-5f6f512cd3a3

Seeking clarification here that, is it possible do that in SB. If yes, the above mentioned API is the right one?  

So, the requirement here is,
1. We should be able to import a content from one SB instance to Another
2. We should be able to import a Collection from one SB instance to Another

Also
is there  any  API to import an ECAR file? 

Please let me how it is possible. 
(currently i am posting in the Q&A, depending on the response i can move to the appropriate categories)

**Comment by Krishnaj20** [comment]
@vinod303 - Can you let us know why do you want to import? You can access content using the public read APIs in other instance as well.

**Reply by kameshbhr** [reply]
@Krishnaj20 I can give the context. We had an exclusive instance of the SB-ED where we created a set of contents. We have shutdown that and moving to a SaaS instance. So we want to move the content from the earlier instance to the new SaaS instance. We have the ECAR files. Would like to know how to import them to the new instance.

**Reply by vinod303** [reply]
@kameshbhr  thanks

@Krishnaj20 , Additionally,  during a recent conversation with a potential adopter, they wanted to know, whether they can make their contents, which they have created  in DIKSHA, available in their SB SaaS instance.

**Reply by pallakartheekreddy** [reply]
@vinod303 Importing `content` from One SB environment to another SB environment, we can use `import` API. But this API supports only contents (All file upload contents & ECAR) and it will not support collections. 

I have a couple of Questions:
1. Which env you are trying this `import` API?
2. What version of Sunbird instance you are using?

@Krishnaj20

**Reply by vinod303** [reply]
@pallakartheekreddy 
1. Env. in which  the import API is executed -  Sunbird SaaS.
2. SB Version there: SB 4.6

**Reply by pallakartheekreddy** [reply]
@vinod303 Can you please check `import` API onboarded as part of release-4.6 in Sunbird SaaS.

I will explain how the import API works for you Knowlg, When you make an import API call with the required request, that API pushes an event to Kafka, and we have async job `content-auto-creator` (Until 4.7 we are using Samza now we are using Flink for this task) take that event form kafka and it will generate the content new instance. So please make sure both `Import API` and `content-auto-creator` job are Woking fine in Sunbird SaaS.

**Reply by kameshbhr** [reply]
@pallakartheekreddy Can you also let us know if there is a way to import/upload an ECAR file?

**Reply by vinod303** [reply]
@pallakartheekreddy , 
checked with the devops. As per their observations
1. import api is enabled 
2. content-auto-creator job is working fine

As suggested by you, we will check the log with their help on monday and get back

**Reply by pallakartheekreddy** [reply]
@kameshbhr @vinod303 We can use the same import API to import ECML content from one instance to the other. But in the import API request `source` you need to pass publicly accessible read API instead of ECAR file. Please refer below for sample:
```"source": "https://dockstaging.sunbirded.org/api/content/v1/read/do_21337600715072307215013"```

**Reply by pallakartheekreddy** [reply]
@vinod303 did you get the logs? 

@kameshbhr Hope your query is now resolved

**Reply by vinod303** [reply]
@pallakartheekreddy  not yet.  Pls keep this open a day or two more pls. I have yet to get the logs.

**Reply by pallakartheekreddy** [reply, marked as answer]
@vinod303 Hope your query is now resolved. Please mark the thread as answered.

**Reply by vinod303** [reply]
@pallakartheekreddy  thanks for the help. I am closing this thread now. The API works now, but there are some other issues with matching values in the framework from source to destination. I will figure that out and create a document for the same publish it!
